### PR TITLE
gh-63283: IDNA prefix should be case insensitive

### PR DESCRIPTION
**Pull request by ZackerySpytz**
Any capitalization of "xn--" should be acceptable for the ACE prefix
(see https://tools.ietf.org/html/rfc3490#section-5).

Co-Authored-By: Pepijn de Vos <pepijndevos@gmail.com>

https://bugs.python.org/issue19083

Automerge-Triggered-By: @asvetlov

* Issue: gh-63283

### CODE REVIEW

**Comment by ZackerySpytz** [comment]
I plan on adding a news entry.

**Review by corona10 — APPROVED** [review]
Looks good to me :)

File: Lib/encodings/idna.py [hunk 1]
```diff
@@ -121,7 +121,7 @@ def ToUnicode(label):
         except UnicodeError:
             raise UnicodeError("Invalid character in IDN label")
     # Step 3: Check for ACE prefix
-    if not label.startswith(ace_prefix):
+    if not label.lower().startswith(ace_prefix):
```
Comment (asvetlov) [inline Lib/encodings/idna.py]: The change is memory and CPU expensive; explicit check for first 3 characters could be much better I believe.

**Review by asvetlov — DISMISSED** [review]
The change looks good; please add NEWs entry.

**Comment by pepijndevos** [comment]
Original patch author chiming in. Happy to see some activity here after close to a decade. Don't think I have the necessary rights to do the NEWS entry though, so I'll just be patient.

**Pull request merged**